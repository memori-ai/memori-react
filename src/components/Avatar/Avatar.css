@media (min-width: 870px) {
    .memori--blob-container {
        position: relative;
        top: 50%;
        width: 100%;
        margin-bottom: auto;
    }

    .memori--blob-container .memori-blob {
        top: 50%;
        left: 50%;
        width: 70%;
        max-width: 300px;
        height: 70%;
        max-height: 300px;
    }

    .memori--blob-container .memori-blob figure {
        width: 100%;
        height: auto;
        aspect-ratio: 1;
    }

    .memori--blob-container .memori-blob .mainDiv {
        overflow: hidden;
        width: 100%;
        height: auto;
        padding-top: 100%;
    }
}

@media (max-width: 870px) {
    .memori--blob-container {
        position: static;
        display: none;
        width: 100%;
    }

    .memori--blob-container .memori-blob {
        position: relative;
    }
}

@media (max-width: 480px) {
    .memori--blob-container {
        display: none;
    }
}

.memori--avatar-wrapper {
    z-index: 100;
    display: flex;
    width: 100%;
    height: 100%;
    align-items: center;
    justify-content: center;
}

.memori--avatar-wrapper.hidden {
    display: none;
}

@media (max-width: 870px) {
    .memori--avatar-wrapper div,
    .memori--avatar-wrapper canvas,
    .memori--avatar-wrapper model-viewer {
        width: 150px !important;
        height: 150px !important;
    }

    .memori--avatar-wrapper model-viewer {
        margin-top: 2rem;
    }

    .memori--avatar-wrapper div {
        top: 1rem;
    }
}

@media (orientation: landscape) and (max-width: 870px) {
    .memori--avatar-wrapper {
        display: none;
    }
}

.memori--avatar-wrapper .avatar-loader figure {
    width: 100%;
    height: 300px;
    margin: 0;
}

.memori--avatar-wrapper .avatar-loader figure img {
    width: 250px;
    height: 250px;
}

@media (max-width: 870px) {
    .memori--avatar-wrapper .avatar-loader figure img {
        width: 150px;
        height: 150px;
    }
}

.memori--avatar-wrapper .avatar-loader figcaption {
    max-height: 50px;
    font-size: 0.8em;
    text-align: center;
}

.memori--avatar-toggle {
    position: absolute;
    z-index: 1001;
    top: 0;
    right: 0;
    display: inline-flex;
    align-items: center;
    text-align: right;
}

@media (min-width: 871px) {
    .memori--avatar-toggle {
        display: none;
    }
}

.memori--avatar-toggle button.memori-button {
    color: var(--memori-primary, #1890ff);
}

.memori--avatar-toggle .memori--avatar-toggle-text {
    font-size: 0.85em;
}

.memori--avatar-link-to-integrations {
    position: absolute;
    z-index: 101;
    top: 0;
    left: 0;
    display: flex;
    width: 100%;
    height: 100%;
    align-items: flex-start;
    justify-content: flex-start;
    padding: 2rem;
}

@media (max-width: 768px) {
    .memori--avatar-link-to-integrations {
        display: none;
    }
}
