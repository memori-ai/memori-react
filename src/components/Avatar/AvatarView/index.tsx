import { CSSProperties, useState } from 'react';
import React, { Suspense } from 'react';
import { Canvas } from '@react-three/fiber';
import {
  OrbitControls,
  SpotLight,
  Environment,
  PerspectiveCamera,
} from '@react-three/drei';
import { isAndroid, isiOS } from '../../../helpers/utils';
import { AvatarView } from './AvatarComponent/avatarComponent';
import PositionControls from './AvatarComponent/positionControls/positionControls';
import { getLocalConfig } from '../../../helpers/configuration';
import Loader from './AvatarComponent/components/loader';

export interface Props {
  url: string;
  sex: 'MALE' | 'FEMALE';
  fallbackImg?: string;
  eyeBlink?: boolean;
  headMovement?: boolean;
  rotateAvatar?: boolean;
  speaking?: boolean;
  style?: CSSProperties;
  fallback?: React.ReactNode;
  halfBody?: boolean;
  loading?: boolean;
  animation?: string;
  showControls?: boolean;
  isZoomed?: boolean;
  chatEmission?: any;
  enablePositionControls?: boolean;
  setEnablePositionControls: (value: boolean) => void;
  isTotem?: boolean;
  setMeshRef?: any;
  stopProcessing: () => void;
  resetVisemeQueue: () => void;
  updateCurrentViseme: (
    currentTime: number
  ) => { name: string; weight: number } | null;
}

const defaultStyles = {
  halfBody: {
    width: '100%',
    height: '100%',
    minHeight: '500px', // Ensure minimum height
    backgroundColor: 'white',
    borderRadius: '100%',
  },
  fullBody: {
    width: '500px',
    height: '500px',
    backgroundColor: 'white',
  },
};

const getLightingComponent = () =>
  isAndroid() || isiOS() ? (
    <SpotLight
      distance={100}
      position={[-0.3, 0.2, 1.25]}
      angle={Math.PI / 2}
      attenuation={5}
      anglePower={5}
    />
  ) : (
    <Environment files="https://raw.githack.com/pmndrs/drei-assets/456060a26bbeb8fdf79326f224b6d99b8bcce736/hdri/venice_sunset_1k.hdr" />
  );

const getCameraSettings = (halfBody: boolean, isZoomed: boolean) => {
  const baseZ = halfBody ? 0.6 : 3;

  return {
    fov: isZoomed ? 44 : 40,
    position: [0, 0, baseZ],
    target: [0, 0, 0],
  };
};

export default function ContainerAvatarView({
  url,
  sex,
  style,
  rotateAvatar,
  eyeBlink,
  headMovement,
  speaking,
  fallback,
  fallbackImg,
  halfBody = true,
  loading,
  animation,
  showControls = false,
  isZoomed,
  chatEmission,
  stopProcessing,
  resetVisemeQueue,
  updateCurrentViseme,
  enablePositionControls,
  setEnablePositionControls,
  isTotem = false,

}: Props) {
  const [cameraZ, setCameraZ] = useState(
    () => getCameraSettings(halfBody, isZoomed || false).position[2]
  );

  const getAvatarHeight = () => {
    // For totem avatars, use stored config value or default to 50
    if (isTotem) {
      const height = getLocalConfig('avatarHeight', 50);
      return height;
    }

    // For half body avatars, use fixed height of 100
    if (halfBody) {
      return 80;
    }

    // For full body avatars, adjust height based on zoom
    return isZoomed ? 16 : 65;
  };

  const getAvatarDepth = () => {
    // For totem avatars, use stored config value or default to 50
    if (isTotem) {
      const depth = getLocalConfig('avatarDepth', 50);
      return depth;
    }

    // For half body avatars, use fixed depth of 50
    if (halfBody) {
      return 50;
    }

    // For full body avatars, adjust depth based on zoom
    return isZoomed ? 5 : 100;
  };

  const [avatarHeight, setAvatarHeight] = useState(getAvatarHeight());
  const [avatarDepth, setAvatarDepth] = useState(getAvatarDepth());

  return (
    <>
          {enablePositionControls && (
        <PositionControls
          avatarHeight={avatarHeight}
          avatarDepth={avatarDepth}
          halfBody={halfBody}
          setAvatarHeight={setAvatarHeight}
          setAvatarDepth={setAvatarDepth}
          setEnablePositionControls={setEnablePositionControls}
        />
      )}
      <Canvas style={style || defaultStyles.fullBody}>
        <PerspectiveCamera
          makeDefault
          position={[0, 0, cameraZ]}
          fov={getCameraSettings(halfBody, isZoomed || false).fov}
        />

        {rotateAvatar && (
          <OrbitControls
            enablePan={false}
            enableZoom={false}
            target={[0, 0, 0]}
          />
        )}

        <Suspense fallback={fallback || <Loader fallbackImg={fallbackImg} />}>
          {getLightingComponent()}

          <AvatarView
            url={url}
            sex={sex}
            showControls={showControls}
            loading={loading || false}
            animation={animation}
            isZoomed={isZoomed || false}
            eyeBlink={eyeBlink || false}
            headMovement={headMovement || false}
            speaking={speaking || false}
            halfBody={halfBody}
            chatEmission={chatEmission}
            updateCurrentViseme={updateCurrentViseme}
            stopProcessing={stopProcessing}
            resetVisemeQueue={resetVisemeQueue}
            setCameraZ={setCameraZ}
            avatarHeight={avatarHeight}
            avatarDepth={avatarDepth}
          />
        </Suspense>
      </Canvas>
    </>
  );
}
