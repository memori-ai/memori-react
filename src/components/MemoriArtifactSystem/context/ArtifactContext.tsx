import React, { createContext, useContext, useState, ReactNode, useCallback, useMemo } from 'react';
import { ArtifactData, ArtifactSystemState } from '../types/artifact.types';

interface ArtifactContextType {
  state: ArtifactSystemState;
  openArtifact: (artifact: ArtifactData) => void;
  closeArtifact: () => void;
  toggleFullscreen: () => void;
}

const ArtifactContext = createContext<ArtifactContextType | null>(null);

export const ArtifactProvider = ({ children }: { children: ReactNode }) => {
  const [state, setState] = useState<ArtifactSystemState>({
    currentArtifact: null,
    isDrawerOpen: false,
    isFullscreen: false,
  });

  const openArtifact = useCallback((artifact: ArtifactData) => {
    setState(prev => {
      // Only update if the artifact is different
      if (prev.currentArtifact?.id === artifact.id && prev.isDrawerOpen) {
        return prev;
      }
      return {
        currentArtifact: artifact,
        isDrawerOpen: true,
        isFullscreen: false,
      };
    });
  }, []);

  const closeArtifact = useCallback(() => {
    setState(prev => {
      // if (!prev.isDrawerOpen) return prev;
      return {
        ...prev,
        currentArtifact: null,
        isDrawerOpen: false,
        isFullscreen: false,
      };
    });
  }, []);

  const toggleFullscreen = useCallback(() => {
    setState(prev => ({
      ...prev,
      isFullscreen: !prev.isFullscreen,
    }));
  }, []);

  const contextValue = useMemo(() => ({
    state,
    openArtifact,
    closeArtifact,
    toggleFullscreen,
  }), [state, openArtifact, closeArtifact, toggleFullscreen]);

  return (
    <ArtifactContext.Provider value={contextValue}>
      {children}
    </ArtifactContext.Provider>
  );
};

export const useArtifact = () => {
  const context = useContext(ArtifactContext);
  if (!context) {
    throw new Error('useArtifact must be used within ArtifactProvider');
  }
  return context;
};
