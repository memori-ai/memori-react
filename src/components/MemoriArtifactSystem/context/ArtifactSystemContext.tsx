/**
 * ArtifactSystemContext
 * React Context for sharing artifact system state and actions across the application
 */

import React, { createContext, useContext, ReactNode } from 'react';
import { 
  ArtifactSystemState, 
  ArtifactSystemActions, 
  ArtifactSystemConfig,
  ArtifactSystemHook 
} from '../types/artifact.types';
import { useArtifactSystem } from '../hooks/useArtifactSystem';

// Context type definition
interface ArtifactSystemContextType extends ArtifactSystemHook {
  // Additional context-specific properties can be added here if needed
}

// Create the context
const ArtifactSystemContext = createContext<ArtifactSystemContextType | null>(null);

// Provider props interface
interface ArtifactSystemProviderProps {
  children: ReactNode;
  config?: Partial<ArtifactSystemConfig>;
}

// Provider component
export const ArtifactSystemProvider: React.FC<ArtifactSystemProviderProps> = ({ 
  children, 
  config = {} 
}) => {
  const artifactSystem = useArtifactSystem(config);

  return (
    <ArtifactSystemContext.Provider value={artifactSystem}>
      {children}
    </ArtifactSystemContext.Provider>
  );
};

// Custom hook to use the artifact system context
export const useArtifactSystemContext = (): ArtifactSystemContextType => {
  const context = useContext(ArtifactSystemContext);
  
  if (!context) {
    throw new Error(
      'useArtifactSystemContext must be used within an ArtifactSystemProvider'
    );
  }
  
  return context;
};

// Export the context for advanced usage
export { ArtifactSystemContext };
