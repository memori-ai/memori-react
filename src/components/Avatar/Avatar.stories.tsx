import React from 'react';
import { Meta, Story } from '@storybook/react';
import { memori, tenant, integration } from '../../mocks/data';
import I18nWrapper from '../../I18nWrapper';
import Avatar, { Props } from './Avatar';

import './Avatar.css';

const meta: Meta = {
  title: 'Widget/Avatar',
  component: Avatar,
  argTypes: {
    avatar3dVisible: {
      control: {
        type: 'boolean',
      },
    },
  },
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const integrationConfig = JSON.parse(integration.customData ?? '{}');

const Template: Story<Props> = args => {
  const [avatar3dVisible, setAvatar3dVisible] = React.useState(
    args.avatar3dVisible
  );

  return (
    <I18nWrapper>
      <div
        style={
          args.integrationConfig?.avatar === 'customglb' ||
          args.integrationConfig?.avatar === 'readyplayerme' ||
          args.integrationConfig?.avatar === 'readyplayerme-full'
            ? {}
            : { marginTop: '20vw' }
        }
      >
        <Avatar
          {...args}
          integrationConfig={
            args.integrationConfig
              ? {
                  ...args.integrationConfig,
                  avatarURL:
                    args.integrationConfig.avatarURL?.split('#')?.[0] +
                    `#${new Date(Date.now()).toISOString()}`,
                }
              : undefined
          }
          avatar3dVisible={avatar3dVisible}
          setAvatar3dVisible={setAvatar3dVisible}
          key={Date.now()}
        />
      </div>
    </I18nWrapper>
  );
};

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {
  memori,
  tenant,
  instruct: false,
  avatar3dVisible: true,
  setAvatar3dVisible: () => {},
  hasUserActivatedSpeak: false,
  isPlayingAudio: false,
};

export const AvatarInBlob = Template.bind({});
AvatarInBlob.args = {
  memori,
  tenant,
  instruct: false,
  avatar3dVisible: true,
  setAvatar3dVisible: () => {},
  hasUserActivatedSpeak: false,
  isPlayingAudio: false,
  integrationConfig: {
    ...integrationConfig,
    avatar: 'userAvatar',
    avatarURL: memori.avatarURL,
  },
};

export const CustomGLBModel = Template.bind({});
CustomGLBModel.args = {
  memori,
  tenant,
  instruct: false,
  avatar3dVisible: true,
  setAvatar3dVisible: () => {},
  hasUserActivatedSpeak: false,
  isPlayingAudio: false,
  integrationConfig: {
    ...integrationConfig,
    avatar: 'customglb',
    avatarURL:
      'https://assets.memori.ai/api/v2/asset/7383f05a-0788-49b0-b9b9-3bfc402c7ddf.glb#1669136149862',
  },
};

export const ReadyPlayerMeAvatar = Template.bind({});
ReadyPlayerMeAvatar.args = {
  memori,
  tenant,
  instruct: false,
  avatar3dVisible: true,
  setAvatar3dVisible: () => {},
  hasUserActivatedSpeak: false,
  isPlayingAudio: false,
  integration,
  integrationConfig,
};

export const FullbodyReadyPlayerMeAvatar = Template.bind({});
FullbodyReadyPlayerMeAvatar.args = {
  memori,
  tenant,
  instruct: false,
  avatar3dVisible: true,
  setAvatar3dVisible: () => {},
  hasUserActivatedSpeak: false,
  isPlayingAudio: false,
  integration,
  integrationConfig: {
    ...integrationConfig,
    avatar: 'readyplayerme-full',
    avatarURL:
      'https://models.readyplayer.me/63b55751f17e295642bf07a2.glb#1669136149862',
  },
};
