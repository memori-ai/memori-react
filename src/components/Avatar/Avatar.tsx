import React, { memo, useEffect, useState } from 'react';
import {
  Integration,
  Memori,
  Tenant,
} from '@memori.ai/memori-api-client/dist/types';
import ErrorBoundary from '../ErrorBoundary/ErrorBoundary';
import Tooltip from '../ui/Tooltip';
import { getResourceUrl } from '../../helpers/media';
import Blob from '../Blob/Blob';
import ModelViewer from '../CustomGLBModelViewer/ModelViewer';
import Button from '../ui/Button';
import { useTranslation } from 'react-i18next';
import Eye from '../icons/Eye';
import EyeInvisible from '../icons/EyeInvisible';
import Edit from '../icons/Edit';
import cx from 'classnames';
import ContainerAvatarView from './AvatarView';
import { useViseme } from '../../context/visemeContext';
import PositionControls from './AvatarView/AvatarComponent/positionControls/positionControls';
import { getLocalConfig } from '../../helpers/configuration';

export interface Props {
  memori: Memori;
  integration?: Integration;
  integrationConfig?: { [key: string]: any };
  tenant?: Tenant;
  instruct?: boolean;
  hasUserActivatedSpeak?: boolean;
  avatar3dVisible?: boolean;
  setAvatar3dVisible: (visible: boolean) => void;
  isPlayingAudio?: boolean;
  loading?: boolean;
  baseUrl?: string;
  apiUrl?: string;
  animation?: string;
  isZoomed?: boolean;
  chatProps?: any;
  enablePositionControls?: boolean;
  setEnablePositionControls: (value: boolean) => void;
  avatarType?: 'blob' | 'avatar3d' | null;
  isTotem?: boolean;
}

const Avatar: React.FC<Props> = ({
  memori,
  integration,
  integrationConfig,
  tenant,
  instruct = false,
  avatar3dVisible = false,
  setAvatar3dVisible,
  hasUserActivatedSpeak = false,
  isPlayingAudio = false,
  loading = false,
  baseUrl,
  apiUrl,
  animation,
  isZoomed = false,
  chatProps,
  avatarType = null,
  enablePositionControls,
  setEnablePositionControls,
  isTotem = false,
}) => {
  const { t } = useTranslation();
  const [isClient, setIsClient] = useState(false);

  const { stopProcessing, updateCurrentViseme, resetVisemeQueue } = useViseme();

  useEffect(() => {
    setIsClient(true);
  }, []);

  // Get the avatar URL, if the avatar is a user avatar, the avatar URL is the user avatar URL, if the avatar is a default avatar, the avatar URL is the default avatar URL
  const getAvatarUrl = () => {
    if (
      integrationConfig?.avatar === 'userAvatar' &&
      memori.avatarURL &&
      memori.avatarURL.length > 0
    ) {
      return getResourceUrl({
        type: 'avatar',
        tenantID: tenant?.id,
        resourceURI: memori.avatarURL,
        baseURL: baseUrl,
        apiURL: apiUrl,
      });
    }
    return undefined;
  };

  const renderAvatar = () => {
    if (
      (integrationConfig?.avatar === 'readyplayerme' ||
        integrationConfig?.avatar === 'readyplayerme-full' ||
        integrationConfig?.avatar === 'customglb' ||
        integrationConfig?.avatar === 'customrpm') &&
      integrationConfig?.avatarURL &&
      (avatarType && avatarType !== 'blob')
    ) {
      return (
        <>
          <div
            className={cx(
              'memori--avatar-wrapper',
              `memori--avatar-${integrationConfig?.avatar || 'default'}`,
              {
                hidden: !avatar3dVisible,
              }
            )}
          >
            {renderAvatarContent()}
          </div>
          {renderAvatarToggle()}
        </>
      );
    }
    return (
      <div className="memori--blob-container">
        {isClient && <Blob speaking={isPlayingAudio} avatar={getAvatarUrl()} />}
      </div>
    );
  };

  const renderAvatarContent = () => {
    if (!isClient) return null;
    if (
      integrationConfig?.avatar === 'readyplayerme' ||
      integrationConfig?.avatar === 'readyplayerme-full' ||
      integrationConfig?.avatar === 'customrpm' 
    ) {
      return (
        <ErrorBoundary
          fallback={
            <div className="memori--blob-container">
              {isClient && (
                <Blob speaking={isPlayingAudio} avatar={getAvatarUrl()} />
              )}
            </div>
          }
        >
          <ContainerAvatarView
            enablePositionControls={enablePositionControls}
            updateCurrentViseme={updateCurrentViseme}
            url={integrationConfig.avatarURL}
            sex={memori.voiceType === 'FEMALE' ? 'FEMALE' : 'MALE'}
            fallbackImg={getAvatarUrl()}
            headMovement
            eyeBlink
            animation={animation}
            halfBody={integrationConfig.avatar === 'readyplayerme'}
            speaking={isPlayingAudio}
            loading={loading}
            style={getAvatarStyle()}
            stopProcessing={stopProcessing}
            resetVisemeQueue={resetVisemeQueue}
            isZoomed={isZoomed}
            isTotem={isTotem}
            chatEmission={chatProps?.dialogState?.emission}
            setEnablePositionControls={setEnablePositionControls}
          />
        </ErrorBoundary>
      );
    }

    if (integrationConfig?.avatar === 'customglb') {
      return (
        <ModelViewer
          poster={getAvatarUrl() || ''}
          src={integrationConfig.avatarURL}
          alt=""
        />
      );
    }

    return null;
  };

  // Render the avatar toggle
  const renderAvatarToggle = () => (
    <div className="memori--avatar-toggle">
      <Button
        ghost
        onClick={() => setAvatar3dVisible(!avatar3dVisible)}
        icon={avatar3dVisible ? <EyeInvisible /> : <Eye />}
      >
        <span className="memori--avatar-toggle-text">
          {avatar3dVisible ? t('hide') : t('show')}
        </span>
      </Button>
    </div>
  );

  const getAvatarStyle = () => {
    if (integrationConfig?.avatar === 'readyplayerme') {
      return {
        width: '100%',
        height: '100%',
        backgroundColor: 'none',
        // borderRadius: '100%',
        boxShadow: 'none',
      };
    }
    return {
      width: '100%',
      height: '100%',
      backgroundColor: 'none',
    };
  };

  const renderIntegrationsLink = () => {
    if (!(instruct && !hasUserActivatedSpeak && memori.isGiver && tenant?.id))
      return null;

    const href = `https://${tenant.id}/${
      memori.culture === 'it-IT' ? 'it' : 'en'
    }/${memori.ownerUserName}/${memori.name}/integrations${
      integration?.integrationID
        ? `?integration=${integration.integrationID}&openAvatarModal=true`
        : ''
    }`;

    return (
      <div className="memori--avatar-link-to-integrations">
        <a
          className="memori-button memori-button--circle memori-button--outlined"
          href={href}
        >
          <Tooltip content={t('widgetgoToIntegrationsToCustomizeAvatar')}>
            <span className="memori-button--icon">
              <Edit />
            </span>
          </Tooltip>
        </a>
      </div>
    );
  };

  return (
    <>
      {renderAvatar()}
      {renderIntegrationsLink()}
    </>
  );
};

export default memo(Avatar, (prevProps, nextProps) => {
  return JSON.stringify(prevProps) === JSON.stringify(nextProps);
});
