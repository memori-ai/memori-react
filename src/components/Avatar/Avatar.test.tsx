import React from 'react';
import { render } from '@testing-library/react';
import { memori, tenant, integration } from '../../mocks/data';
import Avatar from './Avatar';
import { VisemeProvider } from '../../context/visemeContext';
const integrationConfig = JSON.parse(integration.customData ?? '{}');

it('renders defualt Avatar (blob) unchanged', () => {
  const { container } = render(
    <VisemeProvider>
      <Avatar
        memori={memori}
      tenant={tenant}
      isTotem={false}
      setEnablePositionControls={() => {}}
      instruct={false}
      avatar3dVisible={true}
      setAvatar3dVisible={() => {}}
      hasUserActivatedSpeak={false}
        isPlayingAudio={false}
      />
    </VisemeProvider>
  );
  expect(container).toMatchSnapshot();
});

it('renders Avatar with blob and avatar in blob unchanged', () => {
  const { container } = render(
    <VisemeProvider>
      <Avatar
        memori={memori}
        integrationConfig={{
        ...integrationConfig,
        avatar: 'userAvatar',
        avatarURL: memori.avatarURL,
      }}
      tenant={tenant}
      instruct={false}
      isTotem={false}
      setEnablePositionControls={() => {}}
      avatar3dVisible={true}
      setAvatar3dVisible={() => {}}
      hasUserActivatedSpeak={false}
      isPlayingAudio={false}
      />
    </VisemeProvider>
  );
  expect(container).toMatchSnapshot();
});

it('renders Avatar with custom glb model unchanged', () => {
  const { container } = render(
    <VisemeProvider>
      <Avatar
        memori={memori}
        integration={integration}
        isTotem={false}
        setEnablePositionControls={() => {}}
        integrationConfig={{
        ...integrationConfig,
        avatar: 'customglb',
        avatarURL:
          'https://assets.memori.ai/api/v2/asset/7383f05a-0788-49b0-b9b9-3bfc402c7ddf.glb#1669136149862',
      }}
      tenant={tenant}
      instruct={false}
      avatar3dVisible={true}
      setAvatar3dVisible={() => {}}
      hasUserActivatedSpeak={false}
      isPlayingAudio={false}
      />
    </VisemeProvider>
  );
  expect(container).toMatchSnapshot();
});

it('renders Avatar with rpm 3d avatar unchanged', () => {
  const { container } = render( 
    <VisemeProvider>
      <Avatar
        memori={memori}
      integration={integration}
      isTotem={false}
      setEnablePositionControls={() => {}}
      integrationConfig={{
        ...integrationConfig,
        avatar: 'readyplayerme',
        avatarURL:
          'https://assets.memori.ai/api/v2/asset/b791f77c-1a94-4272-829e-eca82fcc62b7.glb',
      }}
      tenant={tenant}
      instruct={false}
      avatar3dVisible={true}
      setAvatar3dVisible={() => {}}
        hasUserActivatedSpeak={false}
        isPlayingAudio={false}
      />
    </VisemeProvider>
  );
  expect(container).toMatchSnapshot();
});
