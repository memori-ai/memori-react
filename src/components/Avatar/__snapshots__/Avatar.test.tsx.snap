// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Avatar with blob and avatar in blob unchanged 1`] = `
<div>
  <div
    class="memori--blob-container"
  >
    <div
      class="memori-blob"
    >
      <figure>
        <img
          alt=""
          role="presentation"
          src="https://aisuru.com/images/aisuru/square_logo.png"
        />
      </figure>
      <div
        class="mainDiv"
      />
      <div
        class="mainDiv"
      />
      <div
        class="mainDiv"
      />
      <div
        class="mainDiv"
      />
      <div
        class="mainDiv"
      />
    </div>
  </div>
</div>
`;

exports[`renders Avatar with custom glb model unchanged 1`] = `
<div>
  <div
    class="memori--blob-container"
  >
    <div
      class="memori-blob"
    >
      <div
        class="mainDiv"
      />
      <div
        class="mainDiv"
      />
      <div
        class="mainDiv"
      />
      <div
        class="mainDiv"
      />
      <div
        class="mainDiv"
      />
    </div>
  </div>
</div>
`;

exports[`renders Avatar with rpm 3d avatar unchanged 1`] = `
<div>
  <div
    class="memori--blob-container"
  >
    <div
      class="memori-blob"
    >
      <div
        class="mainDiv"
      />
      <div
        class="mainDiv"
      />
      <div
        class="mainDiv"
      />
      <div
        class="mainDiv"
      />
      <div
        class="mainDiv"
      />
    </div>
  </div>
</div>
`;

exports[`renders defualt Avatar (blob) unchanged 1`] = `
<div>
  <div
    class="memori--blob-container"
  >
    <div
      class="memori-blob"
    >
      <div
        class="mainDiv"
      />
      <div
        class="mainDiv"
      />
      <div
        class="mainDiv"
      />
      <div
        class="mainDiv"
      />
      <div
        class="mainDiv"
      />
    </div>
  </div>
</div>
`;
